src/expr/src/scalar/func.rs

```diff
@@ -1365,18 +1365,49 @@ fn ascii<'a>(a: Datum<'a>) -> Datum<'a> {
     }
 }
 
+/// A timestamp with only a time component.
+pub trait TimeLike: chrono::Timelike {
```

Comment:
This may not be the best abstraction to be honest, because it doesn't handle truncation or binning, but from what I can tell binning and truncation are not currently supported with the `Time` data type anyway. So it might be best to handle that in a separate issue. Let me know your thoughts.

## Trunc
```
materialize=> select date_trunc('hour', TIME '11:12:42.666');
ERROR:  Cannot call function date_trunc(unknown, time): arguments cannot be implicitly cast to any implementation's parameters; try providing explicit casts

materialize=> SELECT date_trunc('hour', TIMESTAMP '2001-02-16 20:38:40');
     date_trunc      
---------------------
 2001-02-16 20:00:00
(1 row)

postgres=# select date_trunc('hour', TIME '11:12:42.666');
 date_trunc 
------------
 11:00:00
(1 row)
```

## Binning
```
materialize=> SELECT date_bin('15 minutes', TIME '15:44:17', TIME '00:02:30');
ERROR:  Cannot call function date_bin(unknown, time, time): arguments cannot be implicitly cast to any implementation's parameters; try providing explicit casts

materialize=> SELECT date_bin('15 minutes', TIMESTAMP '2020-02-11 15:44:17', TIMESTAMP '2001-01-01 00:02:30');
      date_bin       
---------------------
 2020-02-11 15:32:30
(1 row)
```
(For some reason I can't get `date_bin` to run on my local postgres installation. I think because it was added in v14 but I only have v13).

Reply:
Just downloaded v14 and it looks like `date_bin` isn't supported for `TME`

```
$ psql
psql (14.1 (Ubuntu 14.1-2.pgdg20.04+1))
Type "help" for help.

postgres=# SELECT date_bin('15 minutes', TIME '15:44:17', TIME '00:02:30');
ERROR:  function date_bin(unknown, time without time zone, time without time zone) does not exist
LINE 1: SELECT date_bin('15 minutes', TIME '15:44:17', TIME '00:02:3...
               ^
HINT:  No function matches the given name and argument types. You might need to add explicit type casts.

postgres=# SELECT date_bin('15 minutes', TIMESTAMP '2020-02-11 15:44:17', TIMESTAMP '2001-01-01 00:02:30');
      date_bin       
---------------------
 2020-02-11 15:32:30
(1 row)
```

Postgres docs seem to support this as well: https://www.postgresql.org/docs/devel/functions-datetime.html
> source is a value expression of type timestamp or timestamp with time zone. (Values of type date are cast automatically to timestamp.)

Reply:
Given that this is a re-specification of the existing abstraction, I think it's a-ok.